lib/spack/spack/util/path.py

```diff
@@ -87,6 +87,21 @@ def path_to_os_path(*pths):
     return ret_pths
 
 
+def ensure_legal_path(pth):
```

Comment:
Two requests:

* Can you rename this to `sanititize_file_path` (when I see a function with the prefix `ensure_` it is often a check that doesn't actually produce a result)
* Can you add a docstring mentioning that this formats strings that are intended to be used as paths that will be created by Spack? In particular I think it's worth clarifying we mean "legal" in the sense that it is a valid name for a file on the filesystem

Reply:
Will do in both instances.